File: Python/Program/Clothing.py
'''
Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 2 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
'''

from Product import Product

class Clothing(Product):
    # Private Attributes
    __size = ""
    __material = ""
    __gender = ""
    
    # Constructor
    def __init__(self, idProduct="", name="", brand="", price="", size="", material="", gender=""):
        super().__init__(idProduct, name, brand, price)
        self.__size = size
        self.__material = material
        self.__gender = gender

    # Getter and Setter
    # Getter 
    def getSize(self):
        return self.__size

    def getMaterial(self):
        return self.__material

    def getGender(self):
        return self.__gender

    # Setter 
    def setSize(self, size):
        self.__size = size

    def setMaterial(self, material):
        self.__material = material

    def setGender(self, gender):
        self.__gender = gender

File: PHP/Program/Clothing.php
<?php
require_once 'Product.php';

class Clothing extends Product {
    // Private Attributes
    private $size = "";
    private $material = "";
    private $gender = "";
    
    // Constructor
    public function __construct($idProduct="", $name="", $brand="", $price="", $size="", $material="", $gender="") {
        parent::__construct($idProduct, $name, $brand, $price);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    /*Getter & Setter*/
    // Getter
    public function getSize() {
        return $this->size;
    }

    public function getMaterial() {
        return $this->material;
    }

    public function getGender() {
        return $this->gender;
    }

    // Setter
    public function setSize($size) {
        $this->size = $size;
    }

    public function setMaterial($material) {
        $this->material = $material;
    }

    public function setGender($gender) {
        $this->gender = $gender;
    }
}

?>
